docs/products/kafka/howto/kafka-streams-with-aiven-for-kafka.rst

```diff
@@ -0,0 +1,217 @@
+
+Please make the suggested changes below to the ``KafkaMusicExampleDriver.java`` class file.
+
+::
+
+     // Change (begin)
+     private static final String DEFAULT_BOOTSTRAP_SERVERS = "kafka-20-biz4-a-exercise1.aivencloud.com:17295";
+     private static final String DEFAULT_SCHEMA_REGISTRY_URL = "https://kafka-20-biz4-a-exercise1.aivencloud.com:17298";
+     // Change (end)
+
+   …
+
+       final int restEndpointPort = Integer.parseInt(args[0]);
+       // Change (begin)
+       final String bootstrapServers = args.length > 1 ? args[1] : DEFAULT_BOOTSTRAP_SERVERS;
+       final String schemaRegistryUrl = args.length > 2 ? args[2] : DEFAULT_SCHEMA_REGISTRY_URL;
+       // Change (end)
+       final String restEndpointHostname = args.length > 3 ? args[3] : DEFAULT_REST_ENDPOINT_HOSTNAME;
+
+   …
+
+       streamsConfiguration.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+       // Change (begin)
+   streamsConfiguration.put(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, "SSL");
+   streamsConfiguration.put(SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG, "/path/to/client.truststore.jks");
+   streamsConfiguration.put(SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG, "secret");
+   streamsConfiguration.put(SslConfigs.SSL_KEYSTORE_TYPE_CONFIG, "PKCS12");
+   streamsConfiguration.put(SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG, "/path/to/client.keystore.p12");
+   streamsConfiguration.put(SslConfigs.SSL_KEYSTORE_PASSWORD_CONFIG, "secret");
+   streamsConfiguration.put(SslConfigs.SSL_KEY_PASSWORD_CONFIG, "secret");
+       // Change (end)
+
+   …
+
+       // create and configure the SpecificAvroSerdes required in this example
+       // Change (begin)
+       Map<String, String> serdeConfig = new HashMap<>();
+   serdeConfig.put(AbstractKafkaAvroSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG, schemaRegistryUrl);
+   serdeConfig.put(AbstractKafkaAvroSerDeConfig.BASIC_AUTH_CREDENTIALS_SOURCE, "USER_INFO");
+   serdeConfig.put(AbstractKafkaAvroSerDeConfig.USER_INFO_CONFIG, "avnadmin:schema-reg-password");
+       // Change (end)
+
+   …
+
+*Please substitute appropriate values for Kafka service URL, Schema
+registry URL, avndmin's password (for schema registry) and key and
+truststore files' location on your computer.*
+
+Building the applications
+-------------------------
+
+::
+
+   $ mvn -DskipTests=true clean package
+
+Running the applications
+------------------------
+
+(From Terminal #1) Start the Music example
+
+::
+
+   $ java -cp ./target/kafka-streams-examples-5.0.0-standalone.jar io.confluent.examples.streams.interactivequeries.kafkamusic.KafkaMusicExample 7070
+
+|
+| (From Terminal #2) Start the data driver/generator that streams data
+  to the topics in Kafka
+
+::
+
+   $ java -cp ./target/kafka-streams-examples-5.0.0-standalone.jar io.confluent.examples.streams.interactivequeries.kafkamusic.KafkaMusicExampleDriver
+
+Checking the data
+-----------------
+
+Point the browser to the following links.
+
+`List all running instances of this
```

Comment:
The localhost links can't be clickable - they cause the linkchecker to fail because it can't reach them when running in CI. One option is to include them as a code sample, which would give the user the option to copy.